### PR TITLE
660 support customize parser options

### PR DESCRIPTION
**Pull request by hui-an-yang**
Thank you for your contribution to Taquito.

Before submitting this PR, please make sure:

- [ ] Your code builds cleanly without any errors or warnings
- [ ] You have added unit tests
- [ ] You have added integration tests (if relevant/appropriate)
- [ ] All public methods or types have TypeDoc coverage with a complete description, and ideally an @example
- [ ] You have added or updated corresponding documentation
- [ ] If relevant, you have written a first draft summary describing the change for inclusion in Release Notes. 

## Release Note Draft Snippet

__If relevant, please write a summary of your change that will be suitable for
inclusion in the Release Notes for the next Taquito release.__

### CODE REVIEW

**Comment by hui-an-yang** [comment]
I marked my pr for issue #660 to be in milestone v14.2 and somehow it became a ticket, it's addressed here https://github.com/ecadlabs/taquito/issues/660

File: integration-tests/wallet-michelson-origination.spec.ts [hunk 1]
```diff
@@ -21,4 +23,59 @@ CONFIGS().forEach(({ lib, rpc, setup }) => {
...
+        code: idMichelsonCode,
+        init: idInitData
+      }).send();
+      await op.confirmation()
+      expect(op.opHash).toBeDefined();
+      expect(op.status).toBeDefined();
+      done();
+    });
+
+    it('no parser configures will use MichelCodecParser by default to originate Michelson code and succeeds', async (done) => {
```
Comment (roxaneletourneau) [inline integration-tests/wallet-michelson-origination.spec.ts:66]: configures -> configured

File: integration-tests/wallet-michelson-origination.spec.ts [hunk 2]
```diff
@@ -21,4 +23,59 @@ CONFIGS().forEach(({ lib, rpc, setup }) => {
...
+      // No parserProvider configured will use MichelCodecParser by dafault (the Michelson will be parsed to JSONMichelson)
```
Comment (roxaneletourneau) [inline integration-tests/wallet-michelson-origination.spec.ts:67]: dafault -> default

File: integration-tests/contract-michelson-origination.spec.ts [hunk 3]
```diff
@@ -1,9 +1,61 @@
...
+        balance: "0",
+        code: idMichelsonCode,
+        init: idInitData
+      });
+      await op.confirmation()
+      expect(op.status).toEqual('applied')
+      done();
+    });
+
+    it('no parser configures will use MichelCodecParser by default to originate Michelson code and succeeds', async (done) => {
```
Comment (roxaneletourneau) [inline integration-tests/contract-michelson-origination.spec.ts:46]: configures -> configured

File: integration-tests/contract-michelson-origination.spec.ts [hunk 4]
```diff
@@ -1,9 +1,61 @@
...
+      // No parserProvider configured will use MichelCodecParser by dafault (the Michelson will be parsed to JSONMichelson)
```
Comment (roxaneletourneau) [inline integration-tests/contract-michelson-origination.spec.ts:47]: dafault -> default

**Review by roxaneletourneau — COMMENTED** [review]
Looks good :) 
Just some typos

File: integration-tests/contract-michelson-origination.spec.ts (hunk 4 above)
Comment (hui-an-yang) [reply integration-tests/contract-michelson-origination.spec.ts]: pushed a commit to fix all typos

**Review by roxaneletourneau — APPROVED** [review]

**Pull request merged**